app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -45,48 +67,71 @@ class DroidCvPattern(
         if (ownsMat) {
             mat?.release()
         }
-
-        mat = null
     }
 
-    private fun resize(target: Mat, size: Size) {
+    private fun resize(source: Mat, target: Mat, size: Size) {
         Imgproc.resize(
-            mat, target,
+            source, target,
             CvSize(size.width.toDouble(), size.height.toDouble()),
             0.0, 0.0, Imgproc.INTER_AREA
         )
     }
 
     override fun resize(size: Size): IPattern {
-        val result = Mat()
-        resize(result, size)
-        return DroidCvPattern(result).tag(tag)
+        val resizedMat = Mat()
+        resize(mat, resizedMat, size)
+
+        var resizedAlpha: Mat? = null
+        if (alpha != null) {
+            resizedAlpha = Mat()
+            resize(alpha!!, resizedAlpha, size)
```

Comment:
I can't avoid this with a `let`, since the compiler complains that something could have set alpha to `null` between the null check and the method call.